### PR TITLE
[PIPE-1131] Zip shared object mapping files

### PR DESCRIPTION
**Pull request by Pezzah**
## Goal

Zip shared object mapping files before sending, to reduce transfer time. Also don't process the file as zipping is very effective

This coresponding PR is required to process the zipped files
https://github.com/bugsnag/bugsnag-event-worker/pull/1043

## Design

See ROAD-570 design for reasons for the approach

### CODE REVIEW

File: src/main/groovy/com/bugsnag/android/gradle/BugsnagUploadNdkTask.groovy [hunk 1]
```diff
@@ -155,67 +153,21 @@ class BugsnagUploadNdkTask extends BugsnagMultiPartUploadTask {
...
-            return false
+        byte[] buffer = new byte[1024];
+        int len;
+        while((len=stdout.read(buffer)) != -1){
+            zipStream.write(buffer, 0, len);
         }
-
-        return true
+        zipStream.close()
+        stdout.close()
```
Comment (fractalwrench) [inline src/main/groovy/com/bugsnag/android/gradle/BugsnagUploadNdkTask.groovy]: it'd be good to close this in a finally block (there's an open PR which cleans this up generally across the plugin)

File: src/main/groovy/com/bugsnag/android/gradle/BugsnagUploadNdkTask.groovy [hunk 2]
```diff
@@ -155,67 +153,21 @@ class BugsnagUploadNdkTask extends BugsnagMultiPartUploadTask {
...
-                }
-
-                line = outReader.readLine()
-            }
-
-            writer.close()
-        } catch (IOException e) {
-            project.logger.error("failed to write symbols for " + arch + ": " + e.getMessage())
-            return false
+        byte[] buffer = new byte[1024];
```
Comment (fractalwrench) [inline src/main/groovy/com/bugsnag/android/gradle/BugsnagUploadNdkTask.groovy]: BufferedOutputStream defaults to `8192`, we could probably increase this to speed up IO

File: src/main/groovy/com/bugsnag/android/gradle/BugsnagUploadNdkTask.groovy (hunk 1 above)
Comment (Pezzah) [reply src/main/groovy/com/bugsnag/android/gradle/BugsnagUploadNdkTask.groovy]: changed

File: src/main/groovy/com/bugsnag/android/gradle/BugsnagUploadNdkTask.groovy (hunk 2 above)
Comment (Pezzah) [reply src/main/groovy/com/bugsnag/android/gradle/BugsnagUploadNdkTask.groovy]: changed

File: src/main/groovy/com/bugsnag/android/gradle/BugsnagUploadNdkTask.groovy [hunk 3]
```diff
@@ -122,7 +123,7 @@ class BugsnagUploadNdkTask extends BugsnagMultiPartUploadTask {
                     outputDir.mkdir()
                 }
 
-                File outputFile = new File(outputDir, arch + ".txt")
+                File outputFile = new File(outputDir, arch + ".gzip")
```
Comment (loopj) [inline src/main/groovy/com/bugsnag/android/gradle/BugsnagUploadNdkTask.groovy]: [Nitpick] .gz is the typical file extension for gzipped files

**Pull request merged**